app/src/main/java/com/duckduckgo/app/di/DaggerWorkerFactory.kt

```diff
@@ -52,22 +56,28 @@ class DaggerWorkerFactory(
 
     override fun createWorker(appContext: Context, workerClassName: String, workerParameters: WorkerParameters): ListenableWorker? {
 
-        val workerClass = Class.forName(workerClassName).asSubclass(ListenableWorker::class.java)
-        val constructor = workerClass.getDeclaredConstructor(Context::class.java, WorkerParameters::class.java)
-        val instance = constructor.newInstance(appContext, workerParameters)
+        try {
+            val workerClass = Class.forName(workerClassName).asSubclass(ListenableWorker::class.java)
+            val constructor = workerClass.getDeclaredConstructor(Context::class.java, WorkerParameters::class.java)
+            val instance = constructor.newInstance(appContext, workerParameters)
 
-        when (instance) {
-            is OfflinePixelScheduler.OfflinePixelWorker -> injectOfflinePixelWorker(instance)
-            is DataClearingWorker -> injectDataClearWorker(instance)
-            is ClearDataNotificationWorker -> injectClearDataNotificationWorker(instance)
-            is PrivacyNotificationWorker -> injectPrivacyNotificationWorker(instance)
-            is SearchPromptNotificationWorker -> injectSearchPromptNotificationWorker(instance)
-            is StickySearchNotificationWorker -> injectStickySearchNotificationWorker(instance)
-            is DismissSearchNotificationWorker -> injectDismissSearchNotificationWorker(instance)
-            else -> Timber.i("No injection required for worker $workerClassName")
+            when (instance) {
+                is OfflinePixelScheduler.OfflinePixelWorker -> injectOfflinePixelWorker(instance)
+                is DataClearingWorker -> injectDataClearWorker(instance)
+                is ClearDataNotificationWorker -> injectClearDataNotificationWorker(instance)
+                is PrivacyNotificationWorker -> injectPrivacyNotificationWorker(instance)
+                is SearchPromptNotificationWorker -> injectSearchPromptNotificationWorker(instance)
+                is StickySearchNotificationWorker -> injectStickySearchNotificationWorker(instance)
+                is DismissSearchNotificationWorker -> injectDismissSearchNotificationWorker(instance)
+                else -> Timber.i("No injection required for worker $workerClassName")
+            }
+
+            return instance
+        } catch (exception: Exception){
+            Timber.i("Worker $workerClassName could not be created")
```

Comment:
Could switch to a higher log level. maybe `e` log level. and include the exception too?

`Timber.e(exception, "....")`